### ISSUE TITLE
HTTP 412 error recurred

### ISSUE THREAD

**Issue by mark-chat** [tomlandia/fuckr#203]
As of today, Friday 13th (!) July, the HTTP 412 error has recurred, preventing logon.

I'm sure the usual helpful folk who can detect what needs to change on the files will chime in and I would be grateful if you could help (yest again!)

**Comment by andrewgamez** [comment]
Update your package.json to:
"version": "3.31.1",
"user-agent": "Grindr3/3.31.1.201806042349 (201806042349; iPhone; iOS 11.4.1)",

**Comment by Fuckr16** [comment]
Fuckr stopped working on my Mac at about 10:30 EDT on 7/12/18. I think it has to do with Grindr Version 4.0.0 and its new release. However, I want to thank @2ilent8cho for the excellent job he did on Fuckr for Mac OS X. Yes there were minor kinks here and there but I have no complaints. It was a working and reliable version of Fuckr, one that I haven't seen in almost a year. To me it seemed perfect and I really am appreciative of your hard work.

**Comment by RobbieTechie** [comment]
This will keep occurring as they update new versions on Apple, it wasn't the update to version 4, it was that they force updates more often on Apple devices, and that's the user agent you are using.  When version 4 is released, depending on structure or api changes there will be more issues.

**Comment by mark-chat** [comment]
Thank you guys!
This is SO much appreciated!

**Comment by newbiehere1** [comment]
Hi can anyone explain in plain english how to get past this on mac osx 10.11.6?
I am not a developer so don't understand all the jargon.
Can't login, can't sign up for new account.
I am  not logged in on another device. Thank you

**Comment by 2ilent8cho** [comment]
Have you had it working before @newbiehere1  ? If so i can just post a link to my Mac Fuckr.app with the changes made in it again.

**Comment by newbiehere1** [comment]
@2ilent8cho Thanks for the reply. No I only downloaded the dmg a few hours ago to try it for the first time but had no luck with it. Unable to do anything because that error appears when attempting to login or create an account.

**Comment by 2ilent8cho** [comment]
@newbiehere1  this is my current working fuckr for MacOS with the latest fix applied to the package.json as detailed above by Andrew. https://www.dropbox.com/s/o5zygdpopyq378o/Fuckr.zip?dl=0   I'm not sure, but i think the sign up part stopped working ages ago due to fuckr no longer being maintained by the creator and grindr making changes. So if you have never used it before it may not work for you at all.

**Comment by newbiehere1** [comment]
@2ilent8cho Thank you I will give it a try and let you know how it goes.

**Comment by RobbieTechie** [comment]
Sign up stopped working, partly due to Google doing updates, as this is based off android, but also Grindr adding a captcha.  @newbiehere1 you could create an account with a phone or tablet, or emulator, then delete the app, but not the account, then use  @2ilent8cho Mac version, and bingo, you'll be in business.

**Comment by 2ilent8cho** [comment]
@dcminicoder  I believe fuckr stores chat data and login details under ~/Library/Application Support/Fuckr so if you download mine and put it somewhere else other than your version it should not conflict.  I would take a backup of the  ~/Library/Application Support/Fuckr just incase it goes wrong.

**Comment by newbiehere1** [comment]
Thanks all. No luck with sign up still. 
I will try what  @RobbieTechie suggested of signing up on the app then deleting the app off my phone... cuz I really don't want it on there.

**Comment by paulodias11** [comment]
@2ilent8cho thnk you so much! i can log in, i can receive messages and images and all. but unfortunately i dont see any thumbnails (and the map doesnt show any profiles...) :x also my messages dont seem to be sent...
any tips? (using mac) thnks again everyone!

**Comment by 2ilent8cho** [comment]
@paulodias11 I will boot up one of my Mac's i have not used fuckr on and see how it behaves over the next few days. It works on 2 of my Mac's but i have been doing the fixes as and when needed over the last year so no idea how it behaves on a fresh Mac.

**Comment by RobbieTechie** [comment]
@2ilent8cho Since you are the only person here I know that is actually helping fix this on Mac and sharing, I can help where I can, as I never did make a Hackintosh to release updates.

I looked at your code through Dropbox, from what I can see, what is likely causing the no profiles showing for @paulodias11  in the file fuckr.js, on line 713, you left the filter PhotoOnly set to true.  A while back, grindr made it so if you don't have Xtra and try to use an xtra feature, it would just return an error.  Switching that to default to false, will take care of that, otherwise at least under Windows, it acts like the app is locked up and acts similar to what @paulodias11 is experiencing.

The other issue people may run into using your shared version is with sending images in chat, where they will show in chat, but once clicked they won't show the full size image.

Line 339
`imageHash: imageHash`
needs added to it  imageType: 0, so the whole line looks like this
`imageHash: imageHash , imageType: 0`

@2ilent8cho Thank you for sharing your work
@dcminicoder What details are you trying to add, I may be able to help with that?

**Comment by 2ilent8cho** [comment]
Thanks @RobbieTechie i have made those changes , booted up a fresh install of MacOS on a spare Mac and created an account on a spare iPad, fuckr worked fine on the clean Mac that has never had fuckr on using the fresh account.  This is on 10.13.5 (High Sierra). 

@paulodias11 @newbiehere1  @dcminicoder  here is a link to the updated version Mac version, this *should* work as i have tested it as above, but as @RobbieTechie says you will have to create the account on an iOS/Android device first if you do not have one at the moment.

**Comment by paulodias11** [comment]
you are just awesome! thank you for you help and actually sharing your code! it totally works.
@RobbieTechie and @2ilent8cho you are my fav people now ;)

**Comment by dcminicoder** [comment]
Thanks guys.   I think there are just several profile attributes not showing up.   Mac only has Age Height Weight Position.   Others like HIV status would be good. Would love the Map view to be automatically zoom to a zoomed position. Looks like default is around 25 miles or so and I jsut want to zoom to less than 1 mile form my position.  Not sure where I'd update that.  

    On Thursday, July 19, 2018 5:50 AM, RobbieTechie <notifications@github.com> wrote:
 

 @2ilent8cho Since you are the only person here I know that is actually helping fix this on Mac and sharing, I can help where I can, as I never did make a Hackintosh to release updates.I looked at your code through Dropbox, from what I can see, what is likely causing the no profiles showing for @paulodias11 in the file fuckr.js, on line 713, you left the filter PhotoOnly set to true. A while back, grindr made it so if you don't have Xtra and try to use an xtra feature, it would just return an error. Switching that to default to false, will take care of that, otherwise at least under Windows, it acts like the app is locked up and acts similar to what @paulodias11 is experiencing.The other issue people may run into using your shared version is with sending images in chat, where they will show in chat, but once clicked they won't show the full size image.Line 339
imageHash: imageHash
needs added to it imageType: 0, so the whole line looks like this
imageHash: imageHash , imageType: 0@2ilent8cho Thank you for sharing your work
@dcminicoder What details are you trying to add, I may be able to help with that?—

**Comment by RobbieTechie** [comment]
If I'm understanding the zoom correctly, you want the map view to zoom in closer to the ground?  You can control the zoom level in the profiles.html file which is in the /views folder.  There is a section for the map 

```
<!-- Map -->
    <ng-map ng-if="view == 'map'" center="[{{$storage.location.lat}},{{$storage.location.lon}}]" zoom="10" id='map'>
      <marker ng-repeat="(id, location) in locations"
              position="[{{location.lat}},{{location.lon}}]"
              id="{{id}}"
              on-click="markerClicked($event)">
      </marker>
    </ng-map>
  </div>
</div>
```
You can adjust the zoom level where it says `zoom="10`"

The other part is a little bit tricky, as it interfaced with the managed fields, which use to be downloaded by the app, then they started building those into the app itself.  I'll use the HIV Status field you mentioned.  This will involve the file fuckr.js and the profiles.html in the /views folder again.  Inside the fuckr.js file is a section where Thom added the sexual positions, in the Mac version, it is on line 797, for the next line below this, you can add
`$scope.HIVStatus = ['', "Negative", "Negative, on PrEP", "Positive", "Positive, undetectable"];`

The programmers at grindr bound numbers to string values for the managed fields, so if you did not add the line above, it would return numbers instead of the actual strings, then you'd have to know the meanings or what they equate to.
Ok, inside the profiles.html file, you can add a table section to the profile view to display the header and data,  There is a section near the bottom that shows `<table>` with Age, Height, Weight, Position, you can add a section for the HIV Status by adding 
`<th>HIV Status</th>`
Then the section below, is where you add the data part, here is the part for the sexual positions
`<td>{{sexualPositions[profile.sexualPosition]}}`
So you need to add the 5th section to match up with the HIV Status, and send it through the wording we made with a pipe, so this
`<td>{{HIVStatus[profile.hivStatus]}}</td>`
Make sure to save both of those after editing, if it did not work properly, meaning basically the transforming of the data, typically you get the app to not show the profile at all, or it will show code, similar to this issue
https://github.com/tomlandia/fuckr/issues/190
I know Victor said he would need a debug version for that, but I knew exactly what was going on there, as I rearranged the profile data to that layout, and knew just from the screen shot, why that happened.  That was when some of the brand new profiles showed up bitching about my work, so I just kept my mouth shut.
I didn't go into massive detail, as some of the managed fields for profiles are in odd orders, or not sequential order and have to be looked at individually, and others don't have managed fields tied to them, such as the HIV last tested date, you can pipe that through a date pipe to transform it, for that one, you could add a 
`<th>Last Tested</th>`
for the table header, and then
`{{profile.lastTestedDate | date }}`
To get the date, which in the app, they tie to the month and year.

**Comment by dcminicoder** [comment]
Hey Robbie. Thanks for that.  You are awesome and can't thank you enough for keeping this project alive.  
  Zoom was simple I am surprised I didn't see that sooner.  Happy to say I am  a "Zoom=16" now   :) That will save me 6 clicks every 10 minutes. LOL
No luck with the other feature yet.   Profile page updated just fine with the field, but all fields are blank.  I am going to check a few things and inspect WIN code to see if I have an extra character or syntax issue.    I must be missing something in here. 
    });    return delete $scope.profile;  };  return $scope.sexualPositions = ['', 'Top', 'Bottom', 'Versatile', 'Vers Bottom', 'Vers Top', 'Oral Only']; $scope.hivStatus = ['', 'Negative', 'Negative, on PrEP', 'Positive', 'Positive, undetectable']; };
If you have any thoughts let me know. 

 

    On Saturday, July 21, 2018 6:30 AM, RobbieTechie <notifications@github.com> wrote:
 

 If I'm understanding the zoom correctly, you want the map view to zoom in closer to the ground? You can control the zoom level in the profiles.html file which is in the /views folder. There is a section for the map
    <ng-map ng-if="view == 'map'" center="[{{$storage.location.lat}},{{$storage.location.lon}}]" zoom="10" id='map'>
      <marker ng-repeat="(id, location) in locations"
              position="[{{location.lat}},{{location.lon}}]"
              id="{{id}}"
              on-click="markerClicked($event)">
      </marker>
    </ng-map>
  </div>
</div>
You can adjust the zoom level where it says zoom="10"The other part is a little bit tricky, as it interfaced with the managed fields, which use to be downloaded by the app, then they started building those into the app itself. I'll use the HIV Status field you mentioned. This will involve the file fuckr.js and the profiles.html in the /views folder again. Inside the fuckr.js file is a section where Thom added the sexual positions, in the Mac version, it is on line 797, for the next line below this, you can add
$scope.HIVStatus = ['', "Negative", "Negative, on PrEP", "Positive", "Positive, undetectable"];The programmers at grindr bound numbers to string values for the managed fields, so if you did not add the line above, it would return numbers instead of the actual strings, then you'd have to know the meanings or what they equate to.
Ok, inside the profiles.html file, you can add a table section to the profile view to display the header and data, There is a section near the bottom that shows <table> with Age, Height, Weight, Position, you can add a section for the HIV Status by adding
<th>HIV Status</th>
Then the section below, is where you add the data part, here is the part for the sexual positions
<td>{{sexualPositions[profile.sexualPosition]}}
So you need to add the 5th section to match up with the HIV Status, and send it through the wording we made with a pipe, so this
<td>{{HIVStatus[profile.hivStatus]}}</td>
Make sure to save both of those after editing, if it did not work properly, meaning basically the transforming of the data, typically you get the app to not show the profile at all, or it will show code, similar to this issue
#190
I know Victor said he would need a debug version for that, but I knew exactly what was going on there, as I rearranged the profile data to that layout, and knew just from the screen shot, why that happened. That was when some of the brand new profiles showed up bitching about my work, so I just kept my mouth shut.
I didn't go into massive detail, as some of the managed fields for profiles are in odd orders, or not sequential order and have to be looked at individually, and others don't have managed fields tied to them, such as the HIV last tested date, you can pipe that through a date pipe to transform it, for that one, you could add a
<th>Last Tested</th>
for the table header, and then
{{profile.lastTestedDate | date }}
To get the date, which in the app, they tie to the month and year.—

**Comment by RobbieTechie** [comment]
You're welcome @dcminicoder In the past I've had plenty of times that I've left something off, or added something on accident. If you get stuck and need more help, just let me know

**Comment by nickstacydown** [comment]
Ok so windows is giving me the 412 error and used on two different places signed up on an iPad then downloaded on a Windows 10 so yea what can I do . U and email me or here
